app/controllers/members_controller.rb

```diff
@@ -62,6 +77,11 @@ def graph
     @link_data_str = edges.to_json.html_safe
   end
 
+  def check_email_confirmed!
+    @member = Member.find(params[:id])
```

Comment:
You're probably fetching a `Member` twice and also not failing when the member cannot be found. Add a `get_member` method that runs in a `before` filter that also returns a 404 on failure and assign `@member` in it. It's the "Rails" way.